app/data_loader.py

```diff
@@ -0,0 +1,68 @@
+import json
+from typing import Any
+
+
+class DataLoader:
+    def __init__(self, file_path: str) -> None:
+        self._file_path = file_path
+        self._data: dict[str, Any] = {}
+        self._fuel_price: float = 0.0
+        self._customers: list[dict[str, Any]] = []
+        self._shops: list[dict[str, Any]] = []
+
+        self._load_data()
+
+    @property
+    def fuel_price(self) -> float:
+        return self._fuel_price
+
+    @property
+    def customers(self) -> list[dict[str, Any]]:
+        return self._customers
+
+    @property
+    def shops(self) -> list[dict[str, Any]]:
+        return self._shops
+
+    def _validate_fuel_price(self) -> None:
+        if "FUEL_PRICE" not in self._data:
+            raise KeyError("No FUEL_PRICE provided")
+
+        fuel_price = self._data["FUEL_PRICE"]
+
+        if fuel_price < 0:
```

Comment:
The check `if fuel_price < 0:` assumes that `fuel_price` is a number. It would be safer to first ensure that `fuel_price` is indeed a number (int or float) before comparing it to zero. Consider adding a type check to ensure robustness.